### PR TITLE
Add common utilities

### PR DESCRIPTION
**Pull request by hihigupt**

### CODE REVIEW

File: demos/common/utilities/crypto_using_mbedtls.c [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    {
+        ulRet = 1;
+    }
+    else
+    {
+        ulRet = 0;
+    }
+
+    mbedtls_md_free( &xCtx );
+    *pulBytesCopied = 32;
```
Comment (danewalton) [inline demos/common/utilities/crypto_using_mbedtls.c:55]: I'm assuming this is copied from somewhere else? Interesting they have this set even if line 47 gets hit

File: demos/common/utilities/mbedtls_freertos_port.c [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+ *
+ * @return Number of bytes sent on success; else a negative value.
+ */
+int mbedtls_platform_send( void * ctx,
+                           const unsigned char * buf,
+                           size_t len )
+{
+    SocketHandle socket;
+
+    //configASSERT( ctx != NULL );
```
Comment (danewalton) [inline demos/common/utilities/mbedtls_freertos_port.c:105]: needed?

File: demos/common/utilities/mbedtls_freertos_port.c [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+ *
+ * @return Number of bytes received if successful; Negative value on error.
+ */
+int mbedtls_platform_recv( void * ctx,
+                           unsigned char * buf,
+                           size_t len )
+{
+    SocketHandle socket;
+
+    //configASSERT( ctx != NULL );
```
Comment (danewalton) [inline demos/common/utilities/mbedtls_freertos_port.c:129]: same here

**Review by danewalton — APPROVED** [review]

File: demos/common/utilities/crypto_using_mbedtls.c (hunk 1 above)
Comment (hihigupt) [reply demos/common/utilities/crypto_using_mbedtls.c:55]: fixed

File: demos/common/utilities/mbedtls_freertos_port.c (hunk 2 above)
Comment (hihigupt) [reply demos/common/utilities/mbedtls_freertos_port.c:105]: removed

**Pull request merged**